libraries/libfc/test/test_ordered_diff.cpp

```diff
@@ -7,13 +7,36 @@ using namespace fc;
 
 BOOST_AUTO_TEST_SUITE(ordered_diff_tests)
 
+void print_diff(const auto& diff) {
+   // std::cout << "Remove: [";
+   // for (const auto& e : diff.remove_indexes) {
+   //    std::cout << e << ", ";
+   // }
+   // std::cout << "]\n";
+   // std::cout << "Add: [";
+   // for (const auto& e : diff.insert_indexes) {
+   //    std::cout << e.first << "|" << e.second << ", ";
+   // }
+   // std::cout << "]\n";
+}
+
+// verify only the inserts are in the diff insert_indexes
+void verify_inserted(const auto& diff, const auto& inserts) {
+   BOOST_TEST_REQUIRE(diff.insert_indexes.size() == inserts.size());
+   for (size_t i = 0; i < inserts.size(); ++i) {
+      BOOST_TEST(diff.insert_indexes[i].second == inserts[i]);
+   }
+}
+
 BOOST_AUTO_TEST_CASE(ordered_diff_test) try {
    using namespace std;
 
    { // Basic case
       vector<char>       source = {'a', 'b', 'c', 'd', 'e'};
       vector<char>       target = {'a', 'c', 'e', 'f'};
       auto               result = ordered_diff<char>::diff(source, target);
+      print_diff(result);
+      verify_inserted(result, std::vector{'f'});
```

Comment:
Since the content of ordered_diff affects consensus, I think in addition to checking source and target are the same after apply_diff, all the fields of the diff should be checked to make sure they are not changed by accident.